### DISCUSSION TITLE
4.12

### DISCUSSION THREAD

**Discussion by Schievel1** [Antynea/grub-btrfs#233] [General]
### Script:
* Print version with `-V` or `--version` (#225 #224)
* Add warning that `grub-mkconfig` must be run eat least once before running the script standalone (#219)
* Improve error message when `grub-btrfs.new` contains syntax errors (#230)
* sanitize snapshot comments from `'` to prevent syntax errors (#198 #230)
* fix silent update-grub failure when filesystem of root partition is not supported by grub (thanks @S-trace, #223 #222)

### Daemon:
* Merged solutions for SystemD and OpenRC into one single daemon (#218) that works with timeshift 22.06 and above (#211 #213)
* Added a systemd .service-file for the daemon
* Added configuration file for OpenRC for the daemon (thanks @Megver83, #210)

### Config:
* GRUB_BTRFS_VERSION holds the version and should not be changed by users (but can be by maintainers if they patch something and want to reflect that in the version for example)

### Makefile:
* Improved help
* compress and install manpages
* added some ASCII art

### Readme:
* Added instructions for the new daemon
* Improved usage instructions
* Added instructions for Kali Linux (thanks @Re4son, #197)
* Fix wording, grammar and spelling mistakes (thanks @Semnodime @budswa)
* Better instructions for Fedora (#221)
* Added man-pages for script (grub-btrfs.8) and daemon (grub-btrfsd.8)

**Comment by SysGuides** [comment]
Hello. 

I notice you've made a few updates to the grub-btrfs repository. Is there any plan to bring back the Add-systemd-volatile branch?

I wrote two articles for Fedora 37 based on the Add-systemd-volatile branch. And it worked flawlessly.
[How to Install Fedora 37 with Snapper and Grub-Btrfs for Full System Rollback](https://sysguides.com/install-fedora-37-with-snapper-and-grub-btrfs/)
[How to Install Fedora 37 with LUKS Full Disk Encryption, Snapper, and Grub-Btrfs for Full System Rollback](https://sysguides.com/install-fedora-37-with-luks-full-disk-encryption-snapper-grub-btrfs/)

With this new update and the grub-btrfsd service enabled, snapshots are not added to the GRUB menu automatically. Every time I create a new snapshot in snapper, I have to execute '`grub2-mkconfig -o /boot/grub2/grub.cfg`' command to add snapshots to GRUB menu.

Thanks

**Reply by Schievel1** [reply]
Ouch I thought that's a stale feature branch. 
Sorry about that, will bring it back

**Reply by SysGuides** [reply]
Thanks a-lot 😊

**Comment by SysGuides** [comment]
Thank you very much for bringing back the Add-systemd-volatile branch. 😊🫡

**Reply by Schievel1** [reply]
You are welcome. But couldn't we rather merge that into one branch again? I don't like maintaining two branches so this one is always going to be stale.

**Reply by SysGuides** [reply]
Hello, Schievel1. Sorry for the delay in responding. Does merging make systemd volatile available? 

In Fedora, because the RPM database is in /usr but the DNF database is in /var, I must keep the /var directory within the main / volume to properly rollback. As a result, I must use the kernel parameter systemd.volatile=state, else Fedora wont boot into snapshots.

Also, I just uploaded a video to Youtube. Oops. https://youtu.be/o_FftGkK7iY

Please let me know. I can remove the video, no problem.

**Reply by Schievel1** [reply]
Merging would make systemd volatile available in the master branch, so in the normal release. 
But I read the PR antynea made because of this and it seems like the implementation is not so good. 

So, let's keep things like the are for now.

**Reply by SysGuides** [reply]
Great!!!
